Automattic/wp-calypso
# Regroup following sites and add seen pixels

**Built on top of https://github.com/Automattic/wp-calypso/pull/42355**
#### Changes proposed in this Pull Request

The Reader sidebar was reworked

![New Project (1)](https://user-images.githubusercontent.com/2129455/83263572-829b4880-a1c7-11ea-8bd8-4292e86e1f08.jpg)

There are 3 new sections:
- Automattic, which will have a combined "All" feed and then list all your followed a8c p2s
- P2, which will be available to all users who follow at least a new P2 blog
- Following, which will have a combined "All" feed and then list all you followed non-a8c sites

We load custom `g.gif` tracking pixels to mark content as seen/unseen on the backend
![Markup 2020-05-29 at 4 18 29 PM](https://user-images.githubusercontent.com/2129455/83264024-2dac0200-a1c8-11ea-804e-cfd1d9c59c3e.png)
The following actions would load a tracking pixel
- Mark post as seen/unseen
- Mark combined posts all as seen/unseen
- Mark feed as seen (specific blog or p2 feed)
- Mark section as seen (specific section like Autmattic/All or Following/All)

Since we using pixels instead of API calls, code around those endpoint's logic was removed.

Additional work:
- We show feed favicons in the sidebar list
- Following/All would be the landing page when navigating to Reader
- Sidebar supports dynamic number of organizations like P2 or Automattic, if in the future users could be part of multiple multisite P2 instances
- Code for section unseen counts is in place, can be tested with the other DIFF
- Seen posts will be enabled in staging environments

#### Testing instructions
- Apply D44156-code
- New sidebar items should make sense
- Tracking pixels should be loaded when marking as seen/unseen (look for `g.gif` images loaded in the network tab that have `v=seen_posts` query param)

**If you want to see unseen counts also revert those and apply D44131-code (still work in progress)**
